### PR TITLE
Improved placeholders and font color grids in high contrast mode

### LINKED ISSUE #14907: Issues with accessibility when using high contrast mode in Windows
## 📝 Provide detailed reproduction steps (if any)

1. While using Windows turn on the high contrast mode in the OS's settings
2. Open the [Font family, size, and color demo](https://ckeditor.com/docs/ckeditor5/latest/features/font.html#demo)
3. Open the font color/background color dropdown

### ✔️ Expected result

You should be able to see the different colors for fonts and backgounds in the text as well as in the dropdown.

### ❌ Actual result

All the colors get dropped. Here's a screenshot from Win 11 using the "Night Sky" high contrast preset:

nightsky

And here's one using the "Desert" preset:

desert

One thing to note here is that Tiny MCE also produces the same effect on Windows, so it's possible that it's some upstream issue related to the high contrast mode itself.

Also, macOS's "Invert colours" accessibility option doesn't have this issue.

## 📃 Other details

* Browser: cross-browser
* OS: Windows 10, Windows 11
* First affected CKEditor version: N/A
* Installed CKEditor plugins: N/A

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Feature (utils): Implemented the `env#isMediaForcedColors` property for forced colors detection (e.g. high contrast mode on Windows). See #14907.

Feature (ui): Implemented `ck-media-forced-colors` and `ck-media-default-colors` mixins for detecting the forced colors (e.g. high contrast mode on Windows) mode. See #14907.

Fix (theme-lark): The caret should be visible in a placeholder while in forced colors mode (e.g. high contrast mode on Windows). Improved the look of the placeholders in the forced colors mode. Closes #14907.

Fix (theme-lark): The color grid component should render as a grid of labels in the forced colors mode (e.g. high contrast mode on Windows). Closes #14907.

## Summary by CodeRabbit

- **New Features**
    - Improved color tile functionality in CKEditor by dynamically setting background color based on forced colors mode.
    - Added support for forced colors mode detection in CKEditor environment for enhanced user experience.

- **Bug Fixes**
    - Enhanced color settings in widgets and UI components to better support media forced colors, improving accessibility.

- **Refactor**
    - Updated dependencies related to UI components in CKEditor.

- **Documentation**
    - No visible changes to end-users.

### CODE REVIEW

File: packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/colorgrid/colorgrid.css [hunk 1]
```diff
@@ -20,47 +21,68 @@
 }
 
 .ck.ck-color-grid__tile {
-	width: var(--ck-color-grid-tile-size);
-	height: var(--ck-color-grid-tile-size);
-	min-width: var(--ck-color-grid-tile-size);
-	min-height: var(--ck-color-grid-tile-size);
-	padding: 0;
 	transition: .2s ease box-shadow;
```
Comment (scofalik) [inline packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/colorgrid/colorgrid.css:24]: I understand that this applies to both normal and contrast mode. Do we want to keep transitions for contrast mode?

File: packages/ckeditor5-table/theme/tablecaption.css [hunk 2]
```diff
@@ -20,12 +22,20 @@
 	padding: .6em;
 	font-size: .75em;
 	outline-offset: -1px;
+
+	/* Improve placeholder rendering in high-constrast mode (https://github.com/ckeditor/ckeditor5/issues/14907). */
+	@media (forced-colors: active) {
```
Comment (scofalik) [inline packages/ckeditor5-table/theme/tablecaption.css:27]: Why not using mixin?

**Review by scofalik — APPROVED** [review]
LGTM, two questions in comments.

File: packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/colorgrid/colorgrid.css (hunk 1 above)
Comment (oleq) [reply packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/colorgrid/colorgrid.css:24]: I think these two domains of UX are orthogonal. I see nothing wrong with leaving it as it is.

File: packages/ckeditor5-table/theme/tablecaption.css (hunk 2 above)
Comment (oleq) [reply packages/ckeditor5-table/theme/tablecaption.css]: I missed it.

**Pull request merged**